src/Board.hpp

```diff
@@ -17,5 +18,14 @@ class Board
           int colNumber,
           const Lines& rows,
           const Lines& cols) noexcept;
+    Board(std::string path);
     ~Board();
 };
+
+class InvalidDimensions : public std::invalid_argument
+{
+public:
+    InvalidDimensions()
+        : invalid_argument("Both dimensions must be at least 3")
```

Comment:
There is also a possibility to provide rowNumber or colNumber which is not matching with the number of arrays in rows or cols like:
```
rowNumber: 3
rows: {
  [1],
  [1, 1]
}